### PR TITLE
Feature: Test CI on different envs

### PR DESCRIPTION
**Pull request by blaketarter**
## Description

This PR runs some of the CI jobs on multiple environments to ensure that this repo works for as many different systems as possible. A number of tweaks were made to both the scripts themselves and the CI workflows to fix some issues.

Partially takes care of #365 when merged.

### 🔗 Reference

> _Relevant links (e.g. issue, design, etc.)_

- [#365](https://github.com/cremalab/app-web/issues/365)

### 📋 Acceptance Criteria

> _Checked off by PR **Assignees**_

- [x] Scripts work in ubuntu
- [x] Scripts work in macOS
- [ ] Scripts work in windows

## 🔎 Reviewer Checklist

> _Checked off by PR **Reviewers**_

- [ ] Merge destination is correct
- [ ] Code is correct as understood and conforms to quality standards
- [ ] Tests have been added where appropriate (unit, visual, end-to-end)
- [ ] Acceptance Criteria have been met

---

> ### Roles & Responsibilities
>
> #### 👨‍💻 Assignee
>
> - Initiator of this PR (be sure to set in GitHub UI)
> - Addresses feedback and change requests
> - Merges PR once approved (usually deletes branch unless `develop` or `release`)
>
> #### 👩‍💻 Reviewer
>
> - Invited to review PR by **Assignee** (via GitHub UI)
> - Is expected to complete a review and address followup

### CODE REVIEW

**Comment by blaketarter** [comment]
@roblafeve What are your thoughts on removing the windows runner from the CI and getting all of the other improvements merged in. That way when the dependency blocking us from supporting windows gets resolved we just have to add that OS back into CI?

**Comment by roblafeve** [comment]
@blaketarter yeah, I think that's the right approach. It's out of our control at the moment, so we might as well not get blocked on it. 👍

**Comment by blaketarter** [comment]
When this is ready to merge I will update the required status checks in this repositories settings. I don't want to disrupt automated PRs until that time though.

File: package.json [hunk 1]
```diff
@@ -22,11 +22,13 @@
     "@types/node": "12.12.53",
     "@types/react": "16.9.43",
     "@types/react-dom": "16.9.8",
+    "cross-env": "^7.0.2",
     "cypress": "4.11.0",
     "dependency-cruiser": "9.9.3",
     "eslint-config-prettier": "6.11.0",
     "eslint-config-react-app": "5.2.1",
     "eslint-plugin-cypress": "2.11.1",
+    "eslint-plugin-import": "^2.22.0",
```
Comment (roblafeve) [inline package.json]: Did this dependency end up providing any benefit?

File: package.json [hunk 2]
```diff
@@ -42,36 +44,38 @@
...
   "scripts": {
-    "start": "EXTEND_ESLINT=true react-scripts start",
-    "start:ci": "EXTEND_ESLINT=true npm-run-all build static",
-    "build": "EXTEND_ESLINT=true react-scripts build",
-    "test": "EXTEND_ESLINT=true react-scripts test --testMatch=[ \"**/?(*.)test.ts?(x)\" ]",
+    "start": "cross-env EXTEND_ESLINT=true react-scripts start",
+    "start:ci": "cross-env EXTEND_ESLINT=true run-s build static",
+    "build": "cross-env EXTEND_ESLINT=true react-scripts build",
+    "test": "cross-env EXTEND_ESLINT=true react-scripts test --testMatch=[ \"**/?(*.)test.ts?(x)\" ]",
     "new:component": "cremalab-scripts-web new component",
```
Comment (roblafeve) [inline package.json:50]: We should probably add `cross-env` to the `@cremalab/scripts`—though I suppose we are using `cross-spawn`... hmmm I wonder. It's probably worth adding the strategy matrix to our tests for `@cremalab/scripts` soon.

**Review by roblafeve — DISMISSED** [review]

File: package.json (hunk 1 above)
Comment (blaketarter) [reply package.json]: Ah no, I think I can remove it. Good call. 👍

File: package.json (hunk 2 above)
Comment (blaketarter) [reply package.json:50]: Yeah I think that's the next place for me to make these changes, it's also a good time for me to get familiar with the process for `@cremalab/scripts`. 😅

File: package.json (hunk 1 above)
Comment (blaketarter) [reply package.json]: When I remove this eslint doesn't work any more. Which is weird because it worked before I added this explicit dependency. 😓

Comment (roblafeve) [reply package.json]: Have you done a post uninstall `npm ci`?

Comment (blaketarter) [reply package.json]: Fixed in 09939c8

**Review by roblafeve — APPROVED** [review]

**Pull request merged**